scripts/linters/general_purpose_linter.py

```diff
@@ -697,52 +686,30 @@ def _check_for_mandatory_pattern_in_file(
                     filepath,
                     pattern_list[pattern_found]['message']))
                 summary_messages.append(summary_message)
-                python_utils.PRINT(summary_message)
 
         return failed, summary_messages
 
-    def _check_mandatory_patterns(self):
+    def check_mandatory_patterns(self):
         """This function checks that all files contain the mandatory
         patterns.
         """
-        if self.verbose_mode_enabled:
-            python_utils.PRINT('Starting mandatory patterns check')
-            python_utils.PRINT('----------------------------------------')
-
+        name = 'Mandatory pattern'
         summary_messages = []
         failed = False
-        stdout = sys.stdout
-        with linter_utils.redirect_stdout(stdout):
-            sets_of_patterns_to_match = [
-                MANDATORY_PATTERNS_REGEXP, MANDATORY_PATTERNS_JS_REGEXP]
-            for filepath in self.all_filepaths:
-                for pattern_list in sets_of_patterns_to_match:
-                    failed, mandatory_summary_messages = (
-                        self._check_for_mandatory_pattern_in_file(
-                            pattern_list, filepath, failed))
-                    summary_messages.extend(mandatory_summary_messages)
-
-            if failed:
-                summary_message = (
-                    '%s Mandatory pattern check failed, see errors above for'
-                    'patterns that should be added.' % (
-                        linter_utils.FAILED_MESSAGE_PREFIX))
-            else:
-                summary_message = (
-                    '%s Mandatory pattern check passed' % (
-                        linter_utils.SUCCESS_MESSAGE_PREFIX))
-            python_utils.PRINT(summary_message)
-
-        python_utils.PRINT('')
-
-        summary_messages.append(summary_message)
-        return summary_messages
-
-    def _check_bad_patterns(self):
+        sets_of_patterns_to_match = [
+            MANDATORY_PATTERNS_REGEXP, MANDATORY_PATTERNS_JS_REGEXP]
+        for filepath in self.all_filepaths:
+            for pattern_list in sets_of_patterns_to_match:
+                failed, mandatory_summary_messages = (
+                    self._check_for_mandatory_pattern_in_file(
+                        pattern_list, filepath, failed))
+                summary_messages.extend(mandatory_summary_messages)
+        return concurrent_task_utils.OutputStream(
+            name, failed, summary_messages, summary_messages)
```

Comment:
the second last summary message is used to pass stripped messages in case if the current linter has a stripper method to strip the messages. but in this linter we do not have such method so we are passing the summary_message twice.

Reply:
Can we use the same terminology to pass the messages? Like create full_messages in this check too and assign it equal to summary_messages and pass that? 

I would love to hear other suggestions too. Though, seeing two summary_messages would be really weird for anyone.

Reply:
Done.